include/boot.h

```diff
@@ -56,6 +56,24 @@ typedef struct __packed lz_header {
 } lz_header_t;
 extern lz_header_t lz_header;
 
+#define MLE_UUID0	0x9082ac5a
+#define MLE_UUID1	0x74a7476f
+#define MLE_UUID2	0xa2555c0f
+#define MLE_UUID3	0x42b651cb
+typedef struct __packed mle_header {
+	u32 uuid[4];
+	u32 size;		/* 0x00000034 MLE header size */
+	u32 version;		/* 0x00020002 MLE version 2.2 */
+	u32 sl_stub_entry;	/* Linear entry point of MLE (virt. address) */
+	/* The following fields are used only for Intel TXT */
+	u32 first_page;		/* First valid page of MLE */
+	u32 start_offset;	/* Offset within binary of first byte of MLE */
+	u32 end_offset;		/* Offset within binary of last byte + 1 of MLE */
+	u32 vector;		/* Bit vector of MLE-supported capabilities */
+	u32 cmdline_start;	/* Starting linear address of command line */
+	u32 cmdline_end;	/* Ending linear address of command line */
+} mle_header_t;
```

Comment:
Who's ABI is mle_header?  If Linuxes, shouldn't this be in in boot-params.h ?  If ours, then it probably ought to have its own header rather than being in the middle of a general dumping ground of stuff.

Reply:
@rossphilipson can correct me if i am confusing things, but this is kind of ours but it is for the linux kernel. so i think it could go in either place, but i am leaning towards boot-params.h since this is a boot parameter structure for the kernel.

Reply:
This is the MLE header defined by TXT that has  to be in the MLE image. I believe we are reusing/re-purposing it for SKINIT. It looks like it was moved to linux-params.h which seems reasonable to me.